omnigibson/action_primitives/curobo.py

```diff
@@ -0,0 +1,590 @@
+                successful_paths = result.get_successful_paths()
+                cmd_plan = successful_paths[np.random.choice(len(successful_paths))]
+                return self.path_to_joint_trajectory(path=cmd_plan, joint_names=sim_js_names)
+
+            else:
+                print("CuRobo plan did not converge to a solution.")
+
+    def path_to_joint_trajectory(self, path, joint_names=None):
+        """
+        Converts raw path from motion generator into joint trajectory sequence
+
+        Args:
+            path (JointState): Joint state path to convert into joint trajectory
+            joint_names (None or list): If specified, the individual joints to use when constructing the joint
+                trajectory. If None, will use all joints.
+
+        Returns:
+            torch.tensor: (N, D) joint trajectory tensor
+        """
+        cmd_plan = self.mg.get_full_js(path)
+        # get only joint names that are in both:
+        idx_list = []
+        common_js_names = []
+        jnts_to_idx = {name: i for i, name in enumerate(self.robot.joints.keys())}
+        joint_names = list(self.robot.joints.keys()) if joint_names is None else joint_names
+        for x in joint_names:
+            if x in cmd_plan.joint_names:
+                idx_list.append(jnts_to_idx[x])
+                common_js_names.append(x)
+
+        cmd_plan = cmd_plan.get_ordered_joint_state(common_js_names)
+        return cmd_plan.position
+
+    def convert_q_to_eef_traj(self, traj, return_aa=False):
+        """
+        Converts a joint trajectory @traj into an equivalent trajectory defined by end effector poses
+
+        Args:
+            traj (np.ndarray): (N, D)-shaped joint trajectory
+            return_aa (bool): Whether to return the interpolated orientations in quaternion or axis-angle representation
+
+        Returns:
+            np.ndarray: (N, [6, 7])-shaped array where each entry is is the (x,y,z) position and (x,y,z,w)
+                quaternion (if @return_aa is False) or (ax, ay, az) axis-angle orientation, specified in the robot
+                frame.
+        """
+        # Prune the relevant joints from the trajectory
+        traj = traj[:, self.robot.arm_control_idx[self.robot.default_arm]]
+        n = len(traj)
+
+        # Use forward kinematic solver to compute the EEF link positions
+        positions = np.zeros((n, 3))
+        orientations = np.zeros((n, 4))  # This will be quat initially but we may convert to aa representation
+
+        for i, qpos in enumerate(traj):
+            pose = self._fk.get_link_poses(joint_positions=qpos, link_names=[self.ee_link])
+            positions[i] = pose[self.ee_link][0]
+            orientations[i] = pose[self.ee_link][1]
+
+        # Possibly convert orientations to aa-representation
+        if return_aa:
+            orientations = T.quat2axisangle(orientations)
+
+        return np.concatenate([positions, orientations], axis=-1)
+
```

Comment:
does collision checking work with cloth objects at all?

Reply:
made note for future PR